src/elements/MetricsSelector/MetricsSelector.js

```diff
@@ -0,0 +1,171 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+import React, { Fragment, useMemo, useRef, useState, useEffect, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import { Field, useField } from 'react-final-form'
+import { OnChange } from 'react-final-form-listeners'
+import classNames from 'classnames'
+
+import { PopUpDialog } from 'igz-controls/components'
+import { SelectOption } from 'igz-controls/elements'
+import NameFilter from '../../common/NameFilter/NameFilter'
+import { TextTooltipTemplate, Tooltip } from 'iguazio.dashboard-react-controls/dist/components'
+import { ReactComponent as Caret } from 'igz-controls/images/dropdown.svg'
+import { generateMetricsItems } from './metricsSelector.utils'
+import { METRICS_SELECTOR_OPTIONS } from '../../types'
+
+import './metricsSelector.scss'
+
+const MetricsDropdownSelector = ({ maxSelectionNumber, metrics, name, onSelect }) => {
+  const { input } = useField(name)
+  const [nameFilter, setNameFilter] = useState('')
+  const [isOpen, setIsOpen] = useState(false)
+  const elementRef = useRef()
+
+  const generatedMetrics = useMemo(() => {
+    return generateMetricsItems(metrics, nameFilter, input, maxSelectionNumber)
+  }, [metrics, nameFilter, input, maxSelectionNumber])
+
+  const metricsHeaderClassNames = classNames('metrics-selector-header', isOpen && 'open')
+
+  const clickHandler = useCallback(
+    event => {
+      if (elementRef.current !== event.target.closest('.metrics-selector-popup')) {
+        setIsOpen(false)
+      }
+    },
+    [setIsOpen]
+  )
+
+  const handleScroll = useCallback(
+    event => {
+      if (!event.target.closest('.metrics-selector-options')) {
+        setIsOpen(false)
+      }
+    },
+    [setIsOpen]
+  )
+
+  useEffect(() => {
+    if (isOpen) {
+      window.addEventListener('scroll', handleScroll, true)
+      window.addEventListener('click', clickHandler)
+    }
+
+    return () => {
+      window.removeEventListener('click', clickHandler)
+      window.removeEventListener('scroll', handleScroll, true)
+    }
+  }, [clickHandler, handleScroll, isOpen])
+
+  const handleOnChange = selectedMetrics => {
+    onSelect(selectedMetrics)
+  }
+
+  const getSelectValue = () => {
+    if (!input.value.length) {
+      return 'Select Metrics'
+    }
+    return input.value.length <= 2
+      ? generatedMetrics
+          .filter(metricItem => input.value.includes(metricItem.name))
+          .map(metricItem => metricItem.name)
+          .join(', ')
+      : `${input.value.length} metrics selected`
+  }
+
+  return (
+    <Field name={name}>
```

Comment:
For items that have multiple items the `FieldArray` is used instead of `Field`

Reply:
+